### PR TITLE
ethereum 1.11.2

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Review by p-linnane — DISMISSED** [review]

**Comment by SMillerDev** [comment]
That issue suggests it's no longer needed

**Comment by SMillerDev** [comment]
If Go already defines this according to that linked patch... how is it failing?

**Comment by ZhongRuoyu** [comment]
> If Go already defines this according to that linked patch... how is it failing?

Ah, I had a look at [`logs-ubuntu-22.04`](https://github.com/Homebrew/homebrew-core/suites/11018275879/artifacts/559168659). `-O0` is indeed in the [cgo source](https://github.com/golang/go/blob/master/src/cmd/cgo/gcc.go#L2099), but the superenv removed that.

```
$ sed -En 's/^superenv (.*):.*-O0.*$/\1/p' 01.make.cc | sort | uniq -c
  36 removed
```

In that case, I think `ENV.O0` is still needed.

**Comment by SMillerDev** [comment]
I think @Homebrew/brew should not be filtering this in CGO then since we're introducing a bug here.

**Comment by carlocab** [comment]
Yea, probably. Seems overly conservative, though, since we've been adding our optimisation flags and seen problems in only a handful of formulae that fix it by doing `ENV.O0`.

Still, probably better to play it safe. The tricky part is working out whether the compiler shim was called by `go` or not. An easy solution would be a `go` shim that sets `HOMEBREW_OPTIMIZATION_LEVEL=O0`.

**Comment by SMillerDev** [comment]
Could we add it to `std_go_args`?

**Comment by carlocab** [comment]
> Could we add it to `std_go_args`?

How would that work?

**Comment by SMillerDev** [comment]
I know very little about go 😅. Just trying to help brainstorm this.

**Comment by alebcay** [comment]
> Still, probably better to play it safe. The tricky part is working out whether the compiler shim was called by `go` or not. An easy solution would be a `go` shim that sets `HOMEBREW_OPTIMIZATION_LEVEL=O0`.

Based on [this Stack Overflow post](https://stackoverflow.com/questions/44856124/can-i-change-default-compiler-used-by-cgo), it looks like `go` obeys the compiler selected in `CC`/`CXX`.

See also this excerpt from [the documentation for `cgo`](https://pkg.go.dev/cmd/cgo):
> The default C and C++ compilers may be changed by the CC and CXX environment variables, respectively; those environment variables may include command line options.

**Comment by quentinlesceller** [comment]
geth v1.11.2 just released https://github.com/ethereum/go-ethereum/releases/tag/v1.11.2.

File: Formula/ethereum.rb [hunk 1]
```diff
@@ -25,7 +25,9 @@ class Ethereum < Formula
   depends_on "go" => :build
 
   def install
-    # See https://github.com/golang/go/issues/26487
+    # Force superenv to use -O0 to fix "cgo-dwarf-inference:2:8: error:
+    # enumerator value for '__cgo_enum__0' is not an integer constant".
+    # See discussion in #123337.
```
Comment (SMillerDev) [inline Formula/ethereum.rb:30]:
```suggestion
    # Force superenv to use -O0 to fix "cgo-dwarf-inference:2:8: error:
    # enumerator value for '__cgo_enum__0' is not an integer constant".
    # See discussion https://github.com/Homebrew/brew/issues/14763.
```

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**